Sources/SpeziViews/Model/ViewState.swift

```diff
@@ -10,6 +10,52 @@ import Foundation
 
 
 /// Allows SwiftUI views to keep track of their state and communicate to outside views.
+///
+/// A `ViewState` provides a built-in mechanism for tracking the state of a Spezi UI component.
+/// A view can be in either an `idle`, `processing`, or `error` `ViewState`.
+///
+/// ```swift
+/// import SpeziViews
+/// import SwiftUI
+///
+/// enum ProcessingError: LocalizedError {
+///     case resourceError
+/// }
+///
+/// func doSomeProcessing() throws {
+///     // Function may throw an error
+///     throw ProcessingError.resourceError
+/// }
+///
+/// struct OperationStateTestView: View {
+///     @State private var viewState: ViewState = .idle
+///
+///     var body: some View {
+///         VStack {
+///             Button("Action") {
+///                 viewState = .processing
+///                 do {
+///                     try doSomeProcessing()
+///                     viewState = .idle
+///                 } catch let error as LocalizedError {
+///                     viewState = .error(error)
+///                 } catch {
+///                     print("An unknown error occurred: \(error)")
+///                 }
```

Comment:
I think it is not a good example to show a log that is never processed; I would suggest to point to `AnyLocalizedError` as a nice way to catch localized errors. And a good way to point to this in a `Tip` above when we might cover that error takes a `LocalizedError`?

Reply:
Yes, I wasn't quite happy with that simple `print` either, but leaving out the last `catch` will throw a compiler warning, won't it? Because not every error is a `LocalizedError`. Is there a neat way to avoid that?

Reply:
Oh, nevermind, that's what `AnyLocalizedError` is for. My bad! :)